packages/core/src/routes/experience/classes/experience-interaction.ts

```diff
@@ -483,8 +483,6 @@ export default class ExperienceInteraction {
     const newProfile = await getNewUserProfileFromVerificationRecord(verificationRecord);
     await this.profile.profileValidator.guardProfileUniquenessAcrossUsers(newProfile);
 
-    await this.signInExperienceValidator.guardMandatoryPasswordOnRegister(verificationRecord);
```

Comment:
See the comments below. This guard is no longer needed. The mandatory password validation will be executed ahead, before jumping into this `createNewUser` method.